fluent/fluent.go

```diff
@@ -140,22 +152,24 @@ func newWithDialer(config Config, d dialer) (f *Fluent, err error) {
 		fmt.Fprintf(os.Stderr, "fluent#New: AsyncConnect is now deprecated, please use Async instead")
 		config.Async = config.Async || config.AsyncConnect
 	}
-
-	if config.Async {
-		f = &Fluent{
-			Config:  config,
-			dialer:  d,
-			pending: make(chan *msgToSend, config.BufferLimit),
-		}
-		f.wg.Add(1)
-		go f.run()
-	} else {
-		f = &Fluent{
-			Config: config,
-			dialer: d,
+	f = &Fluent{
+		Config:  config,
+		dialer:  d,
+		pending: make(chan bufferInput, config.BufferLimit),
+		resultPool: sync.Pool{
+			New: func() interface{} {
+				return make(chan error, 1)
+			},
+		},
+	}
+	if !config.Async {
+		if err = f.connect(context.Background()); err != nil {
+			return
 		}
-		err = f.connect()
 	}
+
+	f.wg.Add(1)
+	go f.run()
```

Comment:
Those should be only for Async, right?

Reply:
See https://github.com/fluent/fluent-logger-golang/pull/86/files/5e1b54c24eac3ddbace6a299035fd0c0d9370f7f#r498120175